src/http-api/resources/config.js

```diff
@@ -0,0 +1,111 @@
+const IPFS = require('../../ipfs-core')
+const debug = require('debug')
+const get = require('lodash.get')
+const set = require('lodash.set')
+const log = debug('http-api:config')
+log.error = debug('http-api:config:error')
+
+exports = module.exports
+
+exports.getOrSet = {
+  parseArgs: (request, reply) => {
+    const parseValue = (args) => {
+      if (request.query.bool !== undefined) {
+        args.value = args.value === 'true'
+      } else if (request.query.json !== undefined) {
+        try {
+          args.value = JSON.parse(args.value)
+        } catch (err) {
+          log.error(err)
+          return reply({
+            Message: 'failed to unmarshal json. ' + err,
+            Code: 0
+          }).code(500).takeover()
+        }
+      }
+
+      return reply(args)
+    }
+
+    if (request.query.arg instanceof Array) {
+      return parseValue({
+        key: request.query.arg[0],
+        value: request.query.arg[1]
+      })
+    }
+
+    if (request.params.key) {
+      return parseValue({
+        key: request.params.key,
+        value: request.query.arg
+      })
+    }
+
+    if (!request.query.arg) {
+      return reply("Argument 'key' is required").code(400).takeover()
+    }
+
+    return reply({
+      key: request.query.arg
+    })
+  },
+
+  handler: (request, reply) => {
+    const node = new IPFS()
```

Comment:
look at https://github.com/ipfs/js-ipfs/blob/master/src/http-api/resources/id.js#L1

There is only one IPFS node instance per http-api daemon run